source/simulator/assembly.cc

```diff
@@ -237,10 +239,9 @@ namespace aspect
           face_material_model_outputs(face_quadrature.size(), n_compositional_fields),
           neighbor_face_material_model_inputs(face_quadrature.size(), n_compositional_fields),
           neighbor_face_material_model_outputs(face_quadrature.size(), n_compositional_fields),
-          explicit_material_model_inputs(quadrature.size(), n_compositional_fields),
-          explicit_material_model_outputs(quadrature.size(), n_compositional_fields),
           heating_model_outputs(quadrature.size(), n_compositional_fields),
-          face_heating_model_outputs(quadrature.size(), n_compositional_fields)
+          face_heating_model_outputs(face_quadrature.size(), n_compositional_fields),
+          neighbor_face_heating_model_outputs(face_quadrature.size(), n_compositional_fields)
```

Comment:
We already allocate megabytes of memory here, so it probably doesn't matter much. But for my own curiosity: we probably don'e need these two variables for all assemblers. Are we at a point here where at least in principle we could *ask* whether they are needed and could be initialized to zero length?

Nice bugfix in the first line, by the way!